apis/openstack-swift/src/main/java/org/jclouds/openstack/swift/v1/blobstore/functions/ContentMetaDataToMap.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.openstack.swift.v1.blobstore.functions;
+
+import org.jclouds.io.ContentMetadata;
+import org.jclouds.openstack.swift.v1.domain.SwiftObject;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableMap.Builder;
+
+/**
+ * Updates Content Metadata.
+ *
+ * Required to update the Content Length of Object File as Sum of Content length
+ * of Segments
+ * 
+ * @param contentMetadata
+ * 
+ * @return Map with Content Metadata.
+ */
+
+public class ContentMetaDataToMap {
```

Comment:
Replace Manifest request an argument with Content-metadata

blob.getMetadata().getContentMetadata() return 
[cacheControl=null, contentDisposition=null, contentEncoding=null, contentLanguage=null, contentLength=36701504, contentMD5=null, contentType=application/unknown, expires=null]

But we need Map{content-type=application/unknown}

Static Large Object API also does something similar to this.

Suggestion needed here :(